### ISSUE TITLE
[Feature Request]: add more MADs

### ISSUE THREAD

**Issue by tomtomme** [jasp-stats/jasp-issues#2220]
Still valid for 0.19.3

### Description

- [ ] add 3 more ADs

### Purpose

- learning statistics

**Comment by vandenman** [comment]
Usually, we do not have 3-column layouts, nor do we have layouts where the first few items are omitted from a column. So while the proposal sounds fine to me, it is a bit more tricky to implement than it looks.

**Comment by tomtomme** [comment]
Idea:
Lets go with 2 columns then, the first for categorial vars, the second for metric vars:

Dispersion
----------
Ordinal       |  Metric
-----------------------------
IQR                 Mean MAD
MAD               Mean AD
MAD rob        Variance
Min                 Std Dev
Max                Coeff of Var
Range

**Comment by TarandeepKang** [comment]
Hi @tomtomme I will choose this one as my first to fix. Just to confirm what you want the new statistics in the right-hand column to be, I interpret them as
Mean absolute deviation for the median
Median absolute deviation for the Mean
Mean absolute deviation for the Mean

But as you can tell, I don't think I understand your naming conventions correctly because I'm not convinced that those things exist! I get the impression you have probably previously discussed exactly what these are in some previous issue I can't see. So can you please spell out exactly what these tests are? Then, I will see if they exist in a package (ideally stats). 

Also, as I get the impression this module currently belongs to you @vandenman I trust you don't mind me helping you to make this small change? It is a pleasure to meet you, also!

PS I'll introduce myself a little. I'm aiming to dip myself into the water of software development, and work up to more complicated things in the world of JASP. I am a PhD student (final year) in psychology at the University of Warwick in the UK. I am relatively good with R, but completely new to all this GitHub and GUI stuff. I usually work on psychological trauma, and after having used and appreciated your software for years, and opened a substantial number of issues here, I thought it time to try to help out a bit!

**Comment by tomtomme** [comment]
@TarandeepKang 
In fact I do not exactly know if all these already exist in an R-Pakage. But they certainly exist in theory and in statistics text books :D

So in theory. What is computed?

1. Mean absolute deviation of the median

First compute the median. Then compute the deviatons of all data points from this median. Then compute the mean of those deviations.

2. Median absolute deviation of the Mean

First compute the mean. Then compute the deviatons of all data points from this mean. Then compute the median of those deviations.

3. Mean absolute deviation of the Mean

First compute the mean. Then compute the deviatons of all data points from this mean. Then compute the mean of those deviations.

**Comment by tomtomme** [comment]
And since the andy field book is out soon it also may make sense now to NOT reorder the section but just to append these. Or at least one or two of those three...

**Comment by TarandeepKang** [comment]
I agree that we probably shouldn't change the layout if the book is imminent. I may be betraying my ignorance, but I don't consider myself a newbie to statistics, and while I now understand what these concepts do, I haven't heard of them before; and certainly haven't seen them in software. I have tried looking them up on the Internet, to no avail as well as in my copies of Field/ Howell etc. I wonder, might they be known by another name? 

It really would be good to find implementations in a package. because I don't want my first-ever assignment to be hand-coding something Not least because we reduce reliability and computational reproducibility in that way. That's going to be hard if I  have a name the Internet doesn't recognise! :-)

**Comment by tomtomme** [comment]
I will look tomorrow for my probably german textbooks that mention them. The abreviations and names are not common for sure. The naming for those is a wild clusterfuck mostly :D

The last one is mostly known as average deviation or mittlere Abweichung in german. The others are less common.

**Comment by tomtomme** [comment]
@TarandeepKang 
So I dug up some formulas and functions and sources from textbooks, stackoverflow and wikipedia. See updated post above. I hope this is sufficient?
If you want to start small, just implement 3, because it is the most popular. After that would come 1, only then 2. Number 2 only exists theoretically and is not used in practice as far as I can see. But maybe it should be used more. Also all these MADs could also be computed with the mode instead of the median, but those are even less used in practice.

**Comment by tomtomme** [comment]
Also I think Johnny does also have a say at the moment what should go into thie descriptives module. So:

@JohnnyDoorn 
Would be cool if you could weigh in on this one. Implement all 3 MADs (see above)? If not, which of the 3 should we implement? Should we reorder the section or just append the new ones?

**Comment by JohnnyDoorn** [comment]
Hi all,

First off, awesome that you are looking to contribute in kind @TarandeepKang! I just had a look at our Field book and we do not have a screenshot of the statistics tab of the Descriptives module, so we could rearrange things there a bit. I think this is also fairly high on our priority list, since the current layout of Descriptives does not make a whole lot of sense (e.g., the vague distinction between basic plots and customizable plots). However, what I am more concerned about is to overload the GUI with checkboxes. One of the things that helps JASP be intuitive and user-friendly is that we limit how many options are available, relative to SPSS. I think the MAD statistic and its robust version are already a little bit exotic, but definitely has a place in descriptives; but I am not sure whether we need three additional versions of it. 

One thing that is high on my todo list is to work on a robust stats module - this module could have its own descriptives menu, where there is maybe more room for more of these statistics. I'm not sure when I'll have a somewhat working draft of it though. Maybe something like [this](https://github.com/jasp-stats/jasp-issues/issues/3287) could be a nice starting point in the meantime?

Kind regards,
Johnny

**Comment by EJWagenmakers** [comment]
My two cents:
1. As Johnny indicated, we absolutely need to keep things simple. Yes we can add complex methods, but in separate modules. Yes we can add complex options, but under a separate tab ("Advanced options"). We do not want to slide down the path where we add more and more cool stuff, only to discover at the end that we have created an impenetrable jungle.
2. If we add new functionality, and it is not in a new module, we need to think carefully (a) whether we want to add it there in the first place (is it used often? could it be bundled with similar options and put into a new module?); (b) if we want to add it, where it should be added.
3. To retain coherence and simplicity, any changes to the "core" functionality (the existing set of default methods that are always on the ribbon) need to undergo a "democratic process" where multiple JASP team members agree the change is a good idea.
4. In this particular case, I strongly support Johnny's suggestion to add these metrics to a new "robust stats" module (which I am excited about, by the way)
EJ

**Comment by EJWagenmakers** [comment]
I am looping in Pablo and Don for advice.

**Comment by tomtomme** [comment]
I also just now expanded the list of issues marked with "first project" to include more probably easy ones:

https://github.com/jasp-stats/jasp-issues/issues?q=state%3Aopen%20label%3A%22First%20project%22

But I cannot really say which of those should have priority. Would be good to have advice here, which ones could pic to get the train goin :D

**Comment by shun2wang** [comment]
Good point, But my understanding of the "first project" is: those projects that are easy for new contributors (such as graduate students) to simply understand the overall picture of JASP and use as an entry point to quickly get started. This does not mean the priority level .In fact, we have relatively clear roadmaps for development priorities, such as syntax mode.

**Comment by tomtomme** [comment]
Closing this one since johnny plans work on a robust stats module to maybe include some of this stuff.
Related to 
https://github.com/jasp-stats/jasp-issues/issues/2305
https://github.com/jasp-stats/jasp-issues/issues/1531
https://github.com/jasp-stats/jasp-issues/issues/2769
#450
